corehq/util/metrics/tests/utils.py

```diff
@@ -0,0 +1,22 @@
+from collections import defaultdict
+from contextlib import contextmanager
+
+import mock
+
+
+@contextmanager
+def patch_datadog():
+    def record(fn, name, value, tags=None):
+        key = tuple([
+            name,
+            tuple([tuple(t.split(':')) for t in (tags or [])]),
+        ])
+        stats[key].append(value)
+
+    stats = defaultdict(list)
+    patch = mock.patch("corehq.util.metrics.datadog._datadog_record", new=record)
+    patch.start()
+    try:
+        yield stats
+    finally:
+        patch.stop()
```

Comment:
```suggestion
    with mock.patch("corehq.util.metrics.datadog._datadog_record", new=record):
        yield stats
```

Reply:
Copied from here https://github.com/dimagi/commcare-hq/blob/ebd9d113c0ca9a1bdddcdb2c1fda8a84180404ad/corehq/util/test_utils.py#L591-L608. Was planning to remove the old one once migration is complete.

Updated the new one: 450f211

Reply:
ha! I don't know why I originally wrote it that way. 👍 on eventual removal.